knimin/templates/pm_plate_map.html

```diff
@@ -0,0 +1,358 @@
+        if (i == 0) {
+          if (j == 0) {
+            row_html += "<td width=20></td>";
+          } else {
+            row_html += "<td class='head'>" + j.toString() + "</td>";
+          }
+        } else if (j == 0) {
+          row_html += "<td class='head'>" + String.fromCharCode(64 + i); + "</td>";
+        } else {
+          var well = j.toString() + "_" + i.toString();
+          var sample = {"sample_id": ""};
+          if (well2id.hasOwnProperty(well)) {
+            sample = layout[well2id[well]];
+          }
+          // The sample ID is stored in attributes: data-sample-original (original ID from the database) and data-sample-current (current ID after the last edit).
+          row_html += ("<td><input class='cell', type='text', data-sample-original='" + sample["sample_id"] + "', data-sample-current='" + sample["sample_id"] + "'></td>");
+        }
+      }
+      row.innerHTML = row_html;
+    }
+
+    // Hide leading zeros
+    $("input.cell").each(function() {
+        $(this).val($(this).attr("data-sample-current").replace(/\b0+/g, ''));
+    });
+
+    // Change a well's style when it gets focus
+    $("input.cell").on("focus", function () {
+      var cell = $(this).parent();
+      cell.closest('tr').children('td:first').css("background-color", "darkkhaki");
+      cell.closest('tr').siblings().first().children(':nth-child(' + (cell.index() + 1) + ')').css("background-color", "darkkhaki");
+      $(this).val($(this).attr("data-sample-current"));
+      $(this).select();
+    });
+
+    // Restore a well's style when it loses focus
+    $("input.cell").on("blur", function () {
+      var cell = $(this).parent();
+      cell.closest('tr').children('td:first').css("background-color", "");
+      cell.closest('tr').siblings().first().children(':nth-child(' + (cell.index() + 1) + ')').css("background-color", "");
+      $(this).attr("data-sample-current", $(this).val());
+      $(this).val($(this).attr("data-sample-current").replace(/\b0+/g, ''));
+    });
+
+    // Check if a cell's value has been changed, and apply style accordingly
+    $("input.cell").on("input", function () {
+      var x = $(this).attr("data-sample-original");
+      if (x && $(this).val() != x) {
+        $(this).css("color", "firebrick");
+      } else {
+        $(this).css("color", "black");
+      }
+    });
+
+    // When finishing editing a well, move focus to another well
+    $("input.cell").on("keypress", function (e) {
+      var cc = $(this).parent();  // current cell
+      var tc = "";  // target cell
+      if (e.which == 27) {  // Esc
+        $(this).val($(this).attr("data-sample-current").replace(/\b0+/g, ''));
+        $(this).blur();
+      } else if (e.which == 9) {  // Tab
+        e.preventDefault();
+        if (!e.shiftKey) {  // without Shift: move right
+          tc = cc.next();
+        } else {  // with Shift: move left
+          tc = cc.prev();
+        }
+      } else if (e.which == 13) {  // Enter
```

Comment:
That's a good idea, as we discussed on Wednesday. Once I catch Greg I will ask for his opinion on a number of questions. I guess this wouldn't be hard.

The answer is yes - the cursor will move to the top of the next column.